File: java/FmsItsInterfaceDemo/src/demo/DemoValuesProvider.java
package demo;

import java.util.UUID;
import java.io.File;
import java.io.IOException;

import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.parsers.DocumentBuilder; 
import org.w3c.dom.Document; 
import org.w3c.dom.NodeList;
import org.xml.sax.SAXException;
import org.w3c.dom.Node; 
import org.w3c.dom.Element; 

public class DemoValuesProvider {
	
	public static Vehicle demoVehicle()
    {
		Vehicle vehicle = new Vehicle();
		vehicle.id = UUID.randomUUID().toString();
		vehicle.externalAttributes = new ExternalAttributes();
		vehicle.externalAttributes.iVRIPriorityEnabled = false;
				
		try {
		
			File file = new File("./config/demo_vehicle.xml");
			if(file.exists() && !file.isDirectory()) { 
				DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance(); 
				DocumentBuilder db = dbf.newDocumentBuilder(); 
				Document doc = db.parse(file); 
				doc.getDocumentElement().normalize();
				NodeList nodeList = doc.getElementsByTagName("vehicle"); 
				Node node = nodeList.item(0); 
				Element tElement = (Element)node;
                
				vehicle.name = tElement.getElementsByTagName("name").item(0).getTextContent();
				vehicle.licensePlate = tElement.getElementsByTagName("licenseplate").item(0).getTextContent();
				vehicle.vehicleType = tElement.getElementsByTagName("vehicletype").item(0).getTextContent();
				vehicle.fuel = tElement.getElementsByTagName("fuel").item(0).getTextContent();
			}
			else {
				System.out.println("file does not exists: demo_vehicle.xml");
				
			}
    	} catch (IOException | ParserConfigurationException | SAXException e) {
    		e.printStackTrace();
		}
	
		return vehicle;
    }
	
	public static ITSAccount demoITSAccount()
    {
		ITSAccount account = new ITSAccount();
		
		try {
			File file = new File("./config/demo_settings.xml");
			if(file.exists() && !file.isDirectory()) { 
			DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance(); 
			DocumentBuilder db = dbf.newDocumentBuilder(); 
			Document doc = db.parse(file); 
			doc.getDocumentElement().normalize();
			
			/* rest api settings */
			NodeList restNode = doc.getElementsByTagName("rest"); 
			Node node = restNode.item(0); 
			Element tElement = (Element)node;
                
			account.apiUrl = tElement.getElementsByTagName("endpoint").item(0).getTextContent();
			account.apiSecretName = tElement.getElementsByTagName("name").item(0).getTextContent();
			account.apiSecret = tElement.getElementsByTagName("secret").item(0).getTextContent();
			account.apiMediaType = tElement.getElementsByTagName("type").item(0).getTextContent();
			
			/* websocket settings */
			NodeList wsNodeList = doc.getElementsByTagName("ws"); 
			Node wsNode = wsNodeList.item(0); 
			Element wsElement = (Element) wsNode;
			
			account.wssUrl = wsElement.getElementsByTagName("endpoint").item(0).getTextContent();
			account.socketSecretName = wsElement.getElementsByTagName("name").item(0).getTextContent();
			account.socketSecret = wsElement.getElementsByTagName("secret").item(0).getTextContent();
			}
			else {
				System.out.println("file does not exists: demo_settings.xml");
				
			}
    	} catch (IOException | ParserConfigurationException | SAXException e) {
    		e.printStackTrace();
		}
		return account;
    }
}

File: c#/FMS-ITS Interface Demo/Helpers/DemoValuesProvider.cs
using FMS_ITS_Interface_Demo.Models;
using static FMS_ITS_Interface_Demo.Models.VehicleModel;
using System.Xml;

namespace FMS_ITS_Interface_Demo
{
    public class DemoValuesProvider
    {
        public static Vehicle demoVehicle()
        {
            Vehicle vehicle = new Vehicle();
            vehicle.id = Guid.NewGuid().ToString();
            vehicle.externalAttributes = new ExternalAttributes() { iVRIPriorityEnabled = "false" };
            vehicle.length = new Length() { unit = "m", value = 14.0 };
            vehicle.height = new Height() { unit = "m", value = 4.0 };
            vehicle.width = new Width() { unit = "m", value = 2.6 };
            vehicle.emptyWeight = new Emptyweight() { unit = "kg", value = 12000.0 };

            XmlDocument xDoc = new XmlDocument();

            //load up the xml from the location 
            xDoc.Load("/demo/config/demo_vehicle.xml");
            foreach (XmlNode node in xDoc.DocumentElement.ChildNodes)
            {
                // first node is the url ... have to go to nexted loc node 
                foreach (XmlNode locNode in node)
                {
                    // thereare a couple child nodes here so only take data from node named loc 
                    if (locNode.Name == "name")
                    {
                        vehicle.name = locNode.InnerText;
                    }
                    if (locNode.Name == "licenseplate")
                    {
                        vehicle.licensePlate = locNode.InnerText;
                    }
                    if (locNode.Name == "vehicletype")
                    {
                        vehicle.vehicleType = locNode.InnerText;
                    }
                    if (locNode.Name == "fuel")
                    {
                        vehicle.fuel = locNode.InnerText;
                    }
                }
            }
            return vehicle;
        }


        public static ItsAccount demoITSAccount()
        {
            ItsAccount account = new ItsAccount();

            XmlDocument xDoc = new XmlDocument();
            //load up the xml from the location 
            xDoc.Load("/demo/config/demo_settings.xml");
            foreach (XmlNode node in xDoc.DocumentElement.ChildNodes)
            {
                if (node.Name == "rest")
                {
                    // first node is the url ... have to go to nexted loc node 
                    foreach (XmlNode locNode in node)
                    {
                        // thereare a couple child nodes here so only take data from node named loc 
                        if (locNode.Name == "name")
                        {
                            account.apiSecretName = locNode.InnerText;
                        }
                        if (locNode.Name == "secret")
                        {
                            account.apiSecret = locNode.InnerText;
                        }
                        if (locNode.Name == "type")
                        {
                            account.apiMediaType = locNode.InnerText;
                        }
                        if (locNode.Name == "endpoint")
                        {
                            account.apiUrl = locNode.InnerText;
                        }
                    }
                }
                if (node.Name == "ws")
                {
                    // first node is the url ... have to go to nexted loc node 
                    foreach (XmlNode locNode in node)
                    {
                        // thereare a couple child nodes here so only take data from node named loc 
                        if (locNode.Name == "name")
                        {
                            account.socketSecretName = locNode.InnerText;
                        }
                        if (locNode.Name == "secret")
                        {
                            account.socketSecret = locNode.InnerText;
                        }
                        if (locNode.Name == "endpoint")
                        {
                            account.wssUrl = locNode.InnerText;
                        }
                    }
                }
            }
            return account;
        }
    }
}
